allennlp/tests/common/params_test.py

```diff
@@ -277,3 +277,29 @@ def from_params(cls, params: Params) -> 'A':
                 "a.b.filename": my_file,
                 "a.b.c.c_file": my_other_file
         }
+
+    def test_as_ordered_dict(self):
+        # keyD > keyC > keyE; keyDA > keyDB; Next all other keys alphabetically
+        preference_orders = [["keyD", "keyC", "keyE"], ["keyDA", "keyDB"]]
+        params = Params({"keyC": "valC", "keyB": "valB", "keyA": "valA", "keyE": "valE",
+                         "keyD": {"keyDB": "valDB", "keyDA": "valDA"}})
+        ordered_params_dict = params.as_ordered_dict(preference_orders)
+        ordered_params_dict_str = str(ordered_params_dict)
+        expected_dict_str = str(({'keyD': {'keyDA': 'valDA', 'keyDB': 'valDB'},
```

Comment:
This test should use an `OrderedDict` also, I think.  It currently works, but only because it's relying on an implementation detail of CPython.

Reply:
Dicts are ordered in python and converting to str retains this order. I tried changing orders in expected dict and the test fails.